### PR TITLE
Add first profilerecorder unit tests

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Adding tests to the profile recorder in a series of PRs.
#### Which issue(s) this PR fixes:

None
#### Does this PR have test?
Yes

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: internal/pkg/daemon/profilerecorder/profilerecorder.go [hunk 1]
```diff
@@ -123,17 +124,15 @@ func (r *RecorderReconciler) Setup(
...
 
 	node := &corev1.Node{}
-	if c.Get(
-		context.Background(),
-		client.ObjectKey{Name: os.Getenv(config.NodeNameEnvKey)},
-		node,
-	) != nil {
+	if err := r.ClientGet(
+		c, client.ObjectKey{Name: os.Getenv(config.NodeNameEnvKey)}, node,
+	); err != nil {
```
Comment (saschagrunert) [inline internal/pkg/daemon/profilerecorder/profilerecorder.go:135]: Oops, this actually was a bug. I guess that's why we write those tests…

File: internal/pkg/daemon/profilerecorder/profilerecorder.go [hunk 2]
```diff
@@ -123,17 +124,15 @@ func (r *RecorderReconciler) Setup(
...
 		return errors.Wrap(err, errGetNode)
 	}
```
Comment (saschagrunert) [inline internal/pkg/daemon/profilerecorder/profilerecorder.go:137]: Oops, this actually was a bug. I guess that's why we write those tests…

**Review by JAORMX — APPROVED** [review]

**Pull request merged**